### DISCUSSION TITLE
Virtual "Scroll Bar" for Embla Carousel

### DISCUSSION THREAD

**Discussion by tobimori** [davidjerleke/embla-carousel#602] [Help] [answered]
I recently implemented a design where it makes sense to have a emulated "scrollbar" for Embla Carousel, instead of a simple progress indicator, although I'm unsure where to start with this. Any ideas?

**Comment by davidjerleke** [comment, marked as answer]
Hi @tobimori,

The code below will move the carousel instantly. I'm guessing dragging the scrollbar should scroll the carousel instantly without any smooth animation? As far as I know, that's how default scrollbars work.

```ts
const relativeDistanceToMove =  -30 // Pixels

emblaApi.internalEngine().scrollBody.useDuration(0)
emblaApi.internalEngine().scrollTo.distance(relativeDistanceToMove, false)
``` 

And you can use the `limit` property to calculate correct position based on how much the virtual scrollbar has moved:

```ts
emblaApi.internalEngine().limit.length // length of the scroll track in px 
emblaApi.internalEngine().limit.max // start position of the carousel in px
emblaApi.internalEngine().limit.min // end position of the carousel in px

console.log(emblaApi.internalEngine().limit)
```

As for syncing the scrollbar when the carousel is scrolling based on other input like drag or arrows, just use the scroll event:

```ts
emblaApi.on('scroll', (emblaApi) => {
  emblaApi.internalEngine().location.get() // get the current location in px
}
```

Good luck and if you build something with this, please feel free to share it here. It might be useful for other developers.

Best,
David

**Comment by tobimori** [comment]
I'm still working on this but will share as soon as it's ready, thanks!

**Reply by davidjerleke** [reply]
@tobimori thanks for sharing! The slide width thing you shared is cool 👍.

**Comment by tobimori** [comment]
This is my implementation with a Stimulus controller.
https://gist.github.com/tobimori/f96e97665d7efe3172c5eed2a4e4deee

I'll add a minimal example when I have time.
